discord_slash/utils/manage_components.py

```diff
@@ -167,51 +150,77 @@ def create_select(
     }
 
 
-async def wait_for_component(
-    client: discord.Client, component: typing.Union[dict, str], check=None, timeout=None
-) -> ComponentContext:
+def get_components_ids(component: typing.Union[str, dict, list]) -> typing.Iterator[str]:
     """
-    Waits for a component interaction. Only accepts interactions based on the custom ID of the component, and optionally a check function.
+    Returns generator with 'custom_id' of component or components.
 
-    :param client: The client/bot object.
-    :type client: :class:`discord.Client`
-    :param component: The component dict or custom ID.
-    :type component: Union[dict, str]
-    :param check: Optional check function. Must take a `ComponentContext` as the first parameter.
-    :param timeout: The number of seconds to wait before timing out and raising :exc:`asyncio.TimeoutError`.
-    :raises: :exc:`asyncio.TimeoutError`
+    :param component: Custom ID or component dict (actionrow or button) or list of previous two.
     """
 
-    def _check(ctx):
-        if check and not check(ctx):
-            return False
-        return (
-            component["custom_id"] if isinstance(component, dict) else component
-        ) == ctx.custom_id
-
-    return await client.wait_for("component", check=_check, timeout=timeout)
+    if isinstance(component, str):
+        yield component
+    elif isinstance(component, dict):
+        if component["type"] == ComponentType.actionrow:
+            yield from (comp["custom_id"] for comp in component["components"])
+        else:
+            yield component["custom_id"]
+    elif isinstance(component, list):
+        # Either list of components (actionrows or buttons) or list of ids
+        yield from (comp_id for comp in component for comp_id in get_components_ids(comp))
+    else:
+        raise IncorrectType(
+            f"Unknown component type of {component} ({type(component)}). "
+            f"Expected str, dict or list"
+        )
+
+
+def _get_messages_ids(message: typing.Union[discord.Message, int, list]) -> typing.Iterator[int]:
+    if isinstance(message, int):
+        yield message
+    elif isinstance(message, discord.Message):
+        yield message.id
+    elif isinstance(message, list):
+        yield from (msg_id for msg in message for msg_id in _get_messages_ids(msg))
+    else:
+        raise IncorrectType(
+            f"Unknown component type of {message} ({type(message)}). "
```

Comment:
I feel sure when you `raise` an error exception in Python you're required to give a comma or concatenate the strings.